lib/logstash/filters/jdbc_static.rb

```diff
@@ -0,0 +1,256 @@
+# encoding: utf-8
+require "logstash/filters/base"
+require "logstash/namespace"
+require_relative "util/loader"
+require_relative "util/db_object"
+require_relative "util/repeating_load_runner"
+require_relative "util/lookup_processor"
+
+# This filter can do multiple enhancements to an event in one pass.
+# Define multiple loader sources and multiple lookup targets.
+#
+# [source,ruby]
+
+#
+module LogStash module Filters class JdbcStatic < LogStash::Filters::Base
+  config_name "jdbc_static"
+
+  # Define the set (Hash) of loaders to fetch remote data and create local tables.
+  # the fetched data will be inserted into the local tables. Make sure that the
+  # local table name, columns and datatypes correspond to the shape of the remote data
+  # being fetched.
+  # You can also specify alternate JDBC strings to use if there is more than one remote database
+  # For example:
+  # loaders => [
+  #   {
+  #     "id" => "country_details"
+  #     "query" => "select code, name from WORLD.COUNTRY"
+  #     "max_rows" => 2000
+  #     local_table => "country"
+  #   },
+  #   {
+  #     id => "servers_load"
+  #     query => "select id, ip, name, location from INTERNAL.SERVERS"
+  #     local_table => "servers"
+  #   }
+  # ]
+  # This is optional. You can provide a pre-populated local database server then no initial loaders are needed.
+  config :loaders, :required => true, :default => [], :validate => [LogStash::Filters::Util::Loader]
```

Comment:
Can we remove the above comment? We mark it as `required` and then we say it's optional.

Reply:
done